File: VB/WindowsApplication1/CustomEditor/RepositoryItemExtendedButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory(""), UserRepositoryItem("Register")> _
	Public Class RepositoryItemExtendedButtonEdit
		Inherits RepositoryItemButtonEdit
		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
		End Sub

		Friend Const EditorName As String = "MyButtonEdit"



		Public Shared Sub Register()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(ExtendedButtonEdit), GetType(RepositoryItemExtendedButtonEdit), GetType(ExtendedButtonEditViewInfo), New ButtonEditPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property

	End Class

End Namespace

File: CS/CS/WindowsApplication1/CustomEditor/RepositoryItemExtendedButtonEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1 {
    [System.ComponentModel.DesignerCategory("")]
	[UserRepositoryItem("Register")]
	public class RepositoryItemExtendedButtonEdit : RepositoryItemButtonEdit { 
		static RepositoryItemExtendedButtonEdit() { 
			Register(); 
		}
		public RepositoryItemExtendedButtonEdit() 
        {
        }
		
		internal const string EditorName = "MyButtonEdit";

  

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(ExtendedButtonEdit),
                typeof(RepositoryItemExtendedButtonEdit), typeof(ExtendedButtonEditViewInfo),
                new ButtonEditPainter(), true, null));
        }
		public override string EditorTypeName { 
			get { return EditorName; } 
		}

	}
 
}